internal/services/cognitive/cognitive_account_rai_blocklist_resource.go

```diff
@@ -0,0 +1,216 @@
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.Cognitive.RaiBlocklistsClient
+			accountId, err := raiblocklists.ParseAccountID(model.CognitiveAccountId)
+			if err != nil {
+				return err
+			}
+
+			locks.ByID(accountId.ID())
+			defer locks.UnlockByID(accountId.ID())
+
+			id, err := raiblocklists.ParseRaiBlocklistID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return err
+			}
+
+			properties := existing.Model
+
+			if metadata.ResourceData.HasChange("description") {
+				properties.Properties.Description = pointer.To(model.Description)
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, *id, *properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (c CognitiveRaiBlocklistResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.RaiBlocklistsClient
+
+			id, err := raiblocklists.ParseRaiBlocklistID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+			accountId := raiblocklists.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName)
+
+			locks.ByID(accountId.ID())
+			defer locks.UnlockByID(accountId.ID())
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (c CognitiveRaiBlocklistResource) IDValidationFunc() func(interface{}, string) ([]string, []error) {
+	return raiblocklists.ValidateRaiBlocklistID
+}
+
+func (c CognitiveRaiBlocklistResource) ModelObject() interface{} {
+	return &cognitiveRaiBlocklistModel{}
+}
+
+func (c CognitiveRaiBlocklistResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.RaiBlocklistsClient
+
+			id, err := raiblocklists.ParseRaiBlocklistID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(existing.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			model := existing.Model
+			if model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", id)
+			}
+
+			state := cognitiveRaiBlocklistModel{
+				Name:               id.RaiBlocklistName,
+				CognitiveAccountId: raiblocklists.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName).ID(),
+			}
+
+			if properties := model.Properties; properties != nil {
+				state.Description = pointer.From(properties.Description)
+			}
```

Comment:
We don't error if the model is nil, can we please refactor this to 
```suggestion
			state := cognitiveRaiBlocklistModel{
				Name:               id.RaiBlocklistName,
				CognitiveAccountId: raiblocklists.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName).ID(),
			}
			
			if model := existing.Model; model != nil {
				if props := model.Properties; props != nil {
					state.Description = pointer.From(properties.Description)
				}
			}
```